nistats/second_level_model.py

```diff
@@ -29,6 +29,34 @@
 from .design_matrix import create_second_level_design
 
 
+def _infer_effect_maps(second_level_input, contrast_def):
+    """Deals with the different possibilities of second_level_input"""
+    # Build the design matrix X and list of imgs Y for GLM fit
+    if isinstance(second_level_input, pd.DataFrame):
+        # If a Dataframe was given, we expect contrast_def to be in map_name
+        def _is_contrast_def(x):
+            return x['map_name'] == contrast_def
+        is_con = second_level_input.apply(_is_contrast_def, axis=1)
+        effect_maps = second_level_input[is_con]['effects_map_path'].tolist()
+
+    elif isinstance(second_level_input[0], FirstLevelModel):
+        # Get the first level model maps
+        effect_maps = []
+        for model in second_level_input:
+            effect_map = model.compute_contrast(contrast_def,
+                                                output_type='effect_size')
+            effect_maps.append(effect_map)
+
+    else:
+        effect_maps = second_level_input
```

Comment:
checks already took place in fit, there is no flexibility in what this can be (it will be a niimg list)